src/screens/claimables/ClaimingTransactionClaimable.tsx

```diff
@@ -0,0 +1,204 @@
+  const { isGasReady, isSufficientGas, isValidGas, selectedGasFee, startPollingGasFees, stopPollingGasFees, updateTxFee } = useGas();
+  const { refetch } = useClaimables({ address: accountAddress, currency: nativeCurrency });
+
+  const [baseTxPayload, setBaseTxPayload] = useState<
+    Omit<TransactionClaimableTxPayload, 'gasLimit' | 'maxPriorityFeePerGas' | 'maxFeePerGas'> | undefined
+  >();
+  const [txPayload, setTxPayload] = useState<TransactionClaimableTxPayload | undefined>();
+  const [claimStatus, setClaimStatus] = useState<ClaimStatus>('idle');
+
+  const provider = useMemo(() => getProvider({ chainId: claimable.chainId }), [claimable.chainId]);
+
+  const buildTxPayload = useCallback(async () => {
+    const payload = {
+      value: '0x0' as const,
+      data: claimable.action.data,
+      from: accountAddress,
+      chainId: claimable.chainId,
+      nonce: await getNextNonce({ address: accountAddress, chainId: claimable.chainId }),
+      to: claimable.action.to,
+    };
+
+    setBaseTxPayload(payload);
+  }, [accountAddress, claimable.action.to, claimable.action.data, claimable.chainId, setBaseTxPayload]);
+
+  useEffect(() => {
+    buildTxPayload();
+  }, [buildTxPayload]);
+
+  useEffect(() => {
+    startPollingGasFees();
+    return () => {
+      stopPollingGasFees();
+    };
+  }, [startPollingGasFees, stopPollingGasFees]);
+
+  const estimateGas = useCallback(async () => {
+    if (!baseTxPayload) {
+      logger.error(new RainbowError('[ClaimingTransactionClaimable]: attempted to estimate gas without a tx payload'));
+      return;
+    }
+
+    const gasParams = parseGasParamsForTransaction(selectedGasFee);
+    const updatedTxPayload = { ...baseTxPayload, ...gasParams };
+
+    const gasLimit = await estimateGasWithPadding(updatedTxPayload, null, null, provider);
+
+    if (!gasLimit) {
+      updateTxFee(null, null);
+      logger.error(new RainbowError('[ClaimingTransactionClaimable]: Failed to estimate gas limit'));
+      return;
+    }
+
+    if (needsL1SecurityFeeChains.includes(claimable.chainId)) {
+      const l1SecurityFee = await ethereumUtils.calculateL1FeeOptimism(
+        // @ts-expect-error - type mismatch, but this tx request structure is the same as in SendSheet.js
+        {
+          to: claimable.action.to,
+          from: accountAddress,
+          value: '0x0',
+          data: claimable.action.data,
+        },
+        provider
+      );
+
+      if (!l1SecurityFee) {
+        updateTxFee(null, null);
+        logger.error(new RainbowError('[ClaimingTransactionClaimable]: Failed to calculate L1 security fee'));
+        return;
+      }
+
+      updateTxFee(gasLimit, null, l1SecurityFee);
+    } else {
+      updateTxFee(gasLimit, null);
+    }
+
+    setTxPayload({ ...updatedTxPayload, gasLimit });
+  }, [baseTxPayload, selectedGasFee, provider, claimable.chainId, claimable.action.to, claimable.action.data, updateTxFee, accountAddress]);
```

Comment:
can you wrap this in a try...catch so that gas estimation failures don't cause runtime errors?